lib/constructorio.js

```diff
@@ -423,11 +423,130 @@ ConstructorIO.prototype = {
       method: 'DELETE',
       url: this.makeUrl(`synonym_groups/${group_id}`),
       auth: this.makeAuthToken(),
+      json,
+    }, (error, response) => {
+      handleServerResponse(error, response, callback);
+    });
+  },
+
+  /**
+   * @description Creates a redirect rule.
+   */
+  addRedirectRule(params, callback) {
+    request({
+      method: 'POST',
+      url: this.makeUrl('redirect_rules'),
+      auth: this.makeAuthToken(),
       json: params,
     }, (error, response) => {
       handleServerResponse(error, response, callback);
     });
   },
+
+  /**
+   * @description Retrieves all redirect rules optionally filtered by query or status.
+   */
+  getRedirectRules(params, callback) {
+    const { num_results_per_page, page, query, status } = params;
+    const qsParams = {};
+
+    if (num_results_per_page) {
+      qsParams.num_results_per_page = num_results_per_page;
+    }
+
+    if (page) {
+      qsParams.page = page;
+    }
+
+    if (query) {
+      qsParams.query = query;
+    }
+
+    if (status) {
+      qsParams.status = status;
+    }
+
+    request({
+      method: 'GET',
+      url: this.makeUrl('redirect_rules'),
+      auth: this.makeAuthToken(),
+      qs: qsParams,
+    }, (error, response) => {
+      handleServerResponse(error, response, callback);
+    });
+  },
+
+  /**
+   * @description Retrieves redirect rule for supplied id.
+   */
+  getRedirectRule(params, callback) {
+    const json = clonedeep(params);
+    const { redirect_rule_id } = json;
+    delete json.redirect_rule_id;
+
+    request({
+      method: 'GET',
+      url: this.makeUrl(`redirect_rules/${redirect_rule_id}`),
+      auth: this.makeAuthToken(),
+      json,
+    }, (error, response) => {
+      handleServerResponse(error, response, callback);
+    });
+  },
+
+  /**
+   * @description Completely updates a redirect rule for supplied id.
+   */
+  setRedirectRule(params, callback) {
```

Comment:
Should this be `modifyRedirectRule` to align with other functions and the API docs?  It doesn't exactly make sense, but is consistent.

Reply:
👍